microprofile/WFLY-14277_MicroProfile_JWT_1_2_Upgrade.adoc

```diff
@@ -0,0 +1,98 @@
+this reason it is anticipated this enhancement will be delivered as a set of component upgrades.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14277[WFLY-14277]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1596[EAP7-1596]
+* https://issues.redhat.com/browse/ELY-2068[ELY-2068]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+The component upgrades can be applied exclusively to the https://github.com/wildfly/wildfly[WildFly] project.
+
+=== Other Interested Projects
+
+The implementation for MicroProfile JWT 1.2 is contained within the https://github.com/smallrye/smallrye-jwt[SmallRye JWT] project.
+
+The https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] project contains dependencies on both 
+MicroProfile and SmallRye JWT.  However this update should not require code changes to WildFly Elytron.
+
+== Requirements
+
+=== Hard Requirements
+
+The following Maven artifact dependencies in WildFly will be updated to switch to MicroProfile JWT 1.2:
+ * org.eclipse.microprofile.jwt:microprofile-jwt-auth-api
+ * org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck
+ * io.smallrye:smallrye-jwt
+ 
+Where each of these projects utilises additional dependencies this RFE will review the required dependencies and update WildFly accordingly.
+
+Similar component upgrades will also be applied to the WildFly Elytron project, however code changes are not expected there in relation
+to the upgrades.
+
+The TCK component upgrade will also require a review of the configuration of the TCK to verify the correct set of tests are executed.
+
+The SmallRye JWT project also intends to split the EE Security authentication mechanism into it's own artifact, once this is available our
+dependencies and integration within the subsystem will be updated to make use of the new artifact in addition to the current dependencies.  This
+new artifact is being delivered by the SmallRye JWT project so can be ignored as a new dependency to the application server.
+
+After the relevant component upgrades are in place WildFly will be required to pass the MicroProfile JWT 1.2 TCK.
+
+=== Nice-to-Have Requirements
+
+None, this RFE will not be undertaking enhancements beyond the upgrade to support MicroProfile JWT 1.2.
+
+=== Non-Requirements
+
+No further enhancements to our MicroProfile JWT integration will be undertaken as part of this RFE.
+
+== Test Plan
+
+This RFE will be bringing in the updated MicroProfile JWT 1.2 API and TCK, as we are not changing how we support MicroProfile
+JWT within WildFly no further test changes will be required other than running the appropriate version of the TCK within our
+testsuite.
```

Comment:
@darranl QE also have the `eap-microprofile-test-suite` which will possibly need adjustments for some of the things mentioned here https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/release-notes.asciidoc#release-notes-for-microprofile-jwt-12

Reply:
@darranl I see you added the following tests in the TCK:

* org.eclipse.microprofile.jwt.tck.config.jwe.PrivateKeyAsPEMClasspathTest
* org.eclipse.microprofile.jwt.tck.config.jwe.PrivateKeyAsJWKClasspathTest
* org.eclipse.microprofile.jwt.tck.config.jwe.PrivateKeyAsJWKSClasspathTest

they seem to cover JWE containing a JWS signed with PEM, JWK and JWKS (= signed JWT)

they correspond to coverage for `Support for decrypting JWT tokens which have been encrypted using RSA-OAEP and A256GCM algorithms and contain the claims or inner-signed JWT tokens (#58)` in [Release Notes for MicroProfile JWT 1.2](https://github.com/eclipse/microprofile-jwt-auth/blob/master/spec/src/main/asciidoc/release-notes.asciidoc#release-notes-for-microprofile-jwt-12)

these tests just cover JWE containing `inner-signed JWT tokens` (= signed JWT), they don't cover JWE containing `claims` (= un-signed JWT)

my question is if we should add tests to cover JWE containing un-signed JWT or not ....

Reply:
@tommaso-borgato As this is only a minor update for the specification version my general feeling is the TCK updates should be covering the changes.  There is nothing really changing about our integration into the application server at the moment so it is really if you feel the need to push into something more in depth for any of these areas but as this is core SmallRye JWT behaviour that may be the better location for any testing that goes above and beyond the TCK testing,